tchannel-core/src/main/java/com/uber/tchannel/utils/TChannelUtilities.java

```diff
@@ -38,26 +39,57 @@
 
     public static final ByteBuf emptyByteBuf = Unpooled.EMPTY_BUFFER;
 
+    public static int scoreAddr(NetworkInterface iface, InetAddress addr) {
+        int score = 0;
+
+        if (addr instanceof Inet4Address) {
+            score += 300;
+        }
+
+        try {
+            if (!iface.isLoopback() && !addr.isLoopbackAddress()) {
```

Comment:
1) Don't you want to also filter out or down score the link local addresses i.e. 169.x.x.x ? 
2) And give preference for 10.x.x.x over 172.x.x.x  ?

And what's the expected behavior when we have both eth0/eth1 (which is very likely) ?